x-pack/plugins/security_solution/public/detection_engine/rule_management/components/rule_details/json_diff/diff_view.tsx

```diff
@@ -0,0 +1,264 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React, { useMemo } from 'react';
+import { css, Global } from '@emotion/react';
+import {
+  Diff,
+  useSourceExpansion,
+  useMinCollapsedLines,
+  parseDiff,
+  tokenize,
+} from 'react-diff-view';
+import 'react-diff-view/style/index.css';
+import type { RenderGutter, HunkData, TokenizeOptions, DiffProps } from 'react-diff-view';
+import unidiff from 'unidiff';
+import { useEuiTheme } from '@elastic/eui';
+import { Hunks } from './hunks';
+import { markEdits, DiffMethod } from './mark_edits';
+
+const useExpand = (hunks: HunkData[], oldSource: string, newSource: string) => {
+  const [hunksWithSourceExpanded, expandRange] = useSourceExpansion(hunks, oldSource);
+  const hunksWithMinLinesCollapsed = useMinCollapsedLines(0, hunksWithSourceExpanded, oldSource);
+
+  return {
+    expandRange,
+    hunks: hunksWithMinLinesCollapsed,
+  };
+};
+
+const useTokens = (hunks: HunkData[], diffMethod: DiffMethod, oldSource: string) => {
+  if (!hunks) {
+    return undefined;
+  }
+
+  const options: TokenizeOptions = {
+    oldSource,
+    highlight: false,
+    enhancers: [
+      /*
+        "markEditsBy" is a slightly modified version of "markEdits" enhancer from react-diff-view
+        to enable word-level highlighting.
+      */
+      markEdits(hunks, diffMethod),
+    ],
+  };
+
+  try {
+    /*
+      Synchroniously applies all the enhancers to the hunks and returns an array of tokens.
+      There's also a way to use a web worker to tokenize in a separate thread.
+      Example can be found here: https://github.com/otakustay/react-diff-view/blob/49cebd0958ef323c830395c1a1da601560a71781/site/components/DiffView/index.tsx#L43
+      It didn't work for me right away, but theoretically the possibility is there.
```

Comment:
If we are satisfied with the current performance, transform this comment into a TODO with a ticketm and delete the last line: `It didn't work for me right away, but theoretically the possibility is there.`

Reply:
Makes sense. I'll remove it for now. Let's see if we are satisfied with the perf. Will create a ticket if we aren't.